### ISSUE TITLE
Support for Android

### ISSUE THREAD

**Issue by madhanraj** [microsoft/msquic#4041]
### Describe the feature you'd like supported

I want to know if there is any support and instruction for building MS QUIC for Android.

### Proposed solution

MS QUIC for Smartphones - validation and evaluation in real-time

**Comment by nibanks** [comment]
Yes, MsQuic can work on Android, but we don't have any automation validating it, therefore it's currently not _officially_ supported. While there may be a few gaps, depending on your scenario (likely in the system TLS integration areas), most things should work. Feel free to try it out, and if you find any specific gaps, please open issues for those (and possibly PRs).

**Comment by gganjang** [comment]
I finished my gov project using MsQUIC as transport protocol. Built a static library on linux (ofcourse..) and used it on android mobile app with JNI integration. We didn't fully investigate the performance and efficiency of the app, but it was sufficient to download 2 FHD streaming videos at the same time and decode them.

**Comment by madhanraj** [comment]
With cross-compilation for ARM (or arm64), it works

**Comment by NilsHoyer** [comment]
Can you give me some hints, how I can compile MSQuic to use it I'm my .Net C# project?

**Comment by ManickaP** [comment]
> Can you give me some hints, how I can compile MSQuic to use it in my .Net C# project?

On Android or in general? And you're using System.Net.Quic directly or HTTP/3 via HttpClient?

**Comment by NilsHoyer** [comment]
I want to connect my .Net Apps for Windows and Android via QUIC, not via HTTP. I asked here, since I thought I have to compile MSQuic for Android and in this issue, somebody tried to do this too.

**Comment by madhanraj** [comment]
> I want to connect my .Net Apps for Windows and Android via QUIC, not via HTTP. I asked here, since I thought I have to compile MSQuic for Android and in this issue, somebody tried to do this too.

To build MSQUIC in Android, you can use any cross-compiling tool to generate ARM or ARM64.

**Comment by ManickaP** [comment]
Putting the msquic lib next to your .NET app binaries should do the trick. If the lib can be loaded, it should light up. You can use `QuicConnection.IsSupported` / `QuicListener.IsSupported` to check if it works. Note that we don't officially support this set up, so there might be some unforeseen problems. But if you encounter any issues, feel free to file an issue in [our repo](https://github.com/dotnet/runtime) and we'll try to help.

**Comment by MakesYT** [comment]
> Putting the msquic lib next to your .NET app binaries should do the trick. If the lib can be loaded, it should light up. You can use `QuicConnection.IsSupported` / `QuicListener.IsSupported` to check if it works. Note that we don't officially support this set up, so there might be some unforeseen problems. But if you encounter any issues, feel free to file an issue in [our repo](https://github.com/dotnet/runtime) and we'll try to help.

I tried this operation,'QuicConnection.IsSupported' prompts true, and it can be used normally, but it will be incorrect again after the actual connection.https://github.com/dotnet/runtime/issues/115450

**Comment by guhetier** [comment]
Hi, any update on this? Is it still something you are trying to do?

**Comment by ManickaP** [comment]
We removed dependency on OpenSSL on OSX in https://github.com/dotnet/runtime/pull/117472. Could we do something similar for the Unix helper as well? Thoughts @rzikm @wfurt @vcsjones ?

**Comment by rzikm** [comment]
Unless I misremember, we did this in https://github.com/dotnet/runtime/pull/117472 and https://github.com/dotnet/runtime/pull/117611

**Comment by wfurt** [comment]
Well, you still need OpenSSL. That part is only relevant to certain validation. But that can still help.
